scipy/optimize/tests/test_zeros.py

```diff
@@ -292,6 +292,10 @@ def test_halley_by_name(self):
                             fprime=f_1, fprime2=f_2, xtol=1e-6)
             assert_allclose(f(r.root), 0, atol=1e-6)
 
+    def test_chandrupatla_by_name(self):
+        r"""Invoke chandrupatla through root_scalar()"""
+        pass
```

Comment:
It's OK to hold off on this. 
To be honest... if you wouldn't mind, can we do something a bit unorthodox? Can we start with `_chandrupatla` as a private function? 
That will keep the scope of this PR more manageable. We can have a separate PR to hook it into `root_scalar`.